### ISSUE TITLE
Can you instantiate a Wavesurfer with a MediaElement which is already playing, without interrupting its playback?

### ISSUE THREAD

**Issue by abrahamian** [wavesurfer-js/wavesurfer.js#1704]
If an audio tag is already present on the page, can an instance of Wavesurfer be created to visualize that element's waveform (and subsequently control navigation/playback), but without reloading or otherwise interrupting the playback of the existing element?

My attempts so far have left me with the impression that, using the `MediaElement` backend, calling `.load(mediaElement)` forces the existing media element to reload and restart its playback. Is there any way around this, if the audio is already present and loaded?

Thanks for any help.

**Comment by thijstriemstra** [comment]
@abrahamian looks like this is your problem? https://github.com/katspaugh/wavesurfer.js/blob/master/src/mediaelement.js#L132 Can you comment that (and/or other relevant stuff in that method) out, check what happens?

**Comment by thijstriemstra** [comment]
yo @abrahamian, any feedback?

**Comment by abrahamian** [comment]
@thijstriemstra, I’m still working on this. 

For context: My specific use case relies on another library (cassette) to instantiate the MediaElement and initiate playback. I’ve had some success implementing what I want by  simply commenting out some lines of code (such as the one you referenced, and others that pause/unpause) in my fork of wavesurfer, but only in Chrome. In Firefox and Safari I get bugs that I am still working to understand.

**Comment by vade** [comment]
TL:DR 

I am enabling and disabling audio tracks on a html 5 video Dom element interactively. I cant get wavesurfer to update the audio graph without reloading the source media and having the default tracks enabled. Does wavesurfer let me 'hot reload' the graph based on the state of a media element?

----

Hi all. 

Firstly, thank you for Wavesurfer - its great and ive been able to very quickly do some fun things. 

My current situation is such that I have a video element which contains multiple tracks in the container video mov / mp4. These are 'broadcast masters' from a major streaming provider and contain 20 tracks of AAC discrete mono audio channels. My web app toggles certain channels on and off for a 'quality control' reason (we listen to tracks in isolation).

My intent is to use wavesurfer.js to visualize the audio which is helpful to identify track assignment and encoding issues.

However, wavesurfer doesnt appear to have a facility to update the audio being previewed if the the active audio track of a video element is changed interactively. Is there a way to tell wavesurfer to not 'reload' the media - but to re-build /re-reference the video elements audio context?

Im by no means a web dev - I'm a multimedia engineer normally working on pro desktop apps in c / c++ so im admittedly out of my element - so forgive me if this is trivial.

**Comment by blisst** [comment]
What was the answer to this question since it was apparently "completed?"  

When I use the "media" option for wavesurfer and give it an HTMLAudioElement, it works for the first track, but does not reload the waveform if the `audio.src` changes.

**Comment by katspaugh** [comment]
@blisst you should call `wavesurfer.load` with that src manually, it doesn't watch for attribute updates.
